app/main/menu/submenu-items.js

```diff
@@ -140,8 +141,11 @@ items.reloadApp = {
   click(item, focusedWindow) {
     if (focusedWindow) {
       focusedWindow.webContents.send('reload-window');
-      focusedWindow.webContents.once('did-finish-load', () => {
-        focusedWindow.webContents.send('new-tab');
+
+      electron.ipcMain.once('guest-window-id', (event, id) => {
```

Comment:
No, we don't ceding ID generation to Electron. The `guest-window-id` message is coming from the renderer process: each one generates its own window id. Every time when loading browser windows (when creating them or reloading them), a new id is generated. Session restore on the browser window after a reload is taken care of here.

The fact that we have ipc messaging here is a manifestation of the general problem described in https://github.com/mozilla/tofino/issues/1354 and https://github.com/mozilla/tofino/issues/1414.